# DPLAfest Research Notebook

In [1]:
import datetime
datetime.datetime.utcnow().isoformat()

'2016-02-23T23:01:21.385362'

### First Run of dpla2resp.py on  using DPLA uiuc.json 

*   **Date** 2016-02-18T23:31:30.293867
*   **Program Output**:
<pre>
Starting conversion from DPLA JSON to RESP at 2016-02-18T23:14:11.811667
Finished conversion of DPLA to RESP at 2016-02-18T23:19:13.380267, total time 5.016666666666667 minutes
</pre>
*   **UIUC Resp file size**: 
<pre>
    -a---         2/18/2016   4:19 PM  245817136 output.resp
</pre>

### Second Run of dpla2resp.py on  using DPLA uiuc.json 
*   **Date** 2016-02-18T23:47:21.61342
*   **Program Output**:
<pre>
Starting conversion from DPLA JSON to RESP at 2016-02-18T23:41:13.767339
Finished conversion of DPLA to RESP at 2016-02-18T23:46:48.130425, total time 5.566666666666666 minutes
</pre>
*   **UIUC Resp file size**: 
<pre>
-a---         2/18/2016   4:46 PM  242124486 output.resp
</pre>

### Third Run of dpla2resp.py on  using DPLA uiuc.json 
*   **Date** 
*   **Program Output**:
<pre>
Starting conversion from DPLA JSON to RESP at 
Finished conversion of DPLA to RESP at , total time 5.683333333333334 minutes
</pre>
*   **UIUC Resp file size**: 
<pre>
-a---         2/18/2016   5:40 PM  267945169 output.resp
</pre>
*   **Bulk upload time**:
<pre>
time `cat uiuc.resp | ~/redis-3.2.0-rc3/src/redis-cli --pipe`
All: command not found
real    0m5.051s
user    0m0.355s
sys     0m0.289s
</pre>

Size and memory of Redis Linked Data Fragments Cache:
<pre>
127.0.0.1:6379> DBSIZE
(integer) 10702
127.0.0.1:6379> INFO memory
# Memory
used_memory:60133768
used_memory_human:57.35M
used_memory_rss:68943872
used_memory_rss_human:65.75M
used_memory_peak:60133768
used_memory_peak_human:57.35M
total_system_memory:2099204096
total_system_memory_human:1.96G
used_memory_lua:37888
used_memory_lua_human:37.00K
maxmemory:0
maxmemory_human:0B
maxmemory_policy:volatile-lru
mem_fragmentation_ratio:1.15
mem_allocator:jemalloc-4.0.3

</pre>
(**NOTE** If running dpla2resp.py in Windows, don't forget to run the Linux utility `dos2unix`) 

### First Run of dpla2resp.py on  using DPLA getty.json 
*   **Date** 2016-02-23T23:01:21.385362
*   **Program Output**:
<pre>
Starting conversion from DPLA JSON to RESP at 2016-02-23T22:57:00.135235
Finised onversion of DPLA to RESP at 2016-02-23T23:36:37.650250, total time 39.61666666666667 minutes
</pre>
*   **Getty Resp file size**: 
<pre>
-a---         2/23/2016   4:36 PM 1104372723 output.resp
</pre>
*   **Bulk upload time**:
<pre>
127.0.0.1:6379> DBSIZE
(integer) 20193
127.0.0.1:6379> INFO memory
# Memory
used_memory:32935264
used_memory_human:31.41M
used_memory_rss:41291776
used_memory_rss_human:39.38M
used_memory_peak:59846840
used_memory_peak_human:57.07M
total_system_memory:2099204096
total_system_memory_human:1.96G
used_memory_lua:37888
used_memory_lua_human:37.00K
maxmemory:0
maxmemory_human:0B
maxmemory_policy:volatile-lru
mem_fragmentation_ratio:1.25
mem_allocator:jemalloc-4.0.3
</pre>